src/GapicClientTrait.php

```diff
@@ -300,7 +302,8 @@ private function setClientOptions(array $options)
                 $options['apiEndpoint'],
                 $transport,
                 $options['transportConfig'],
-                $options['clientCertSource']
+                $options['clientCertSource'],
+                $hasEmulator
```

Comment:
nit: just pass the above assertion here, no need for a local variable:
```suggestion
                $options['hasEmulator'] ?? false
```